include/zephyr/audio/dmic.h

```diff
@@ -28,6 +29,8 @@
  * @brief Abstraction for digital microphones
  *
  * @defgroup audio_dmic_interface Digital Microphone Interface
+ * @since 1.13.0
+ * @version 0.1.0
```

Comment:
IMHO, API versions are redundant. Zephyr is used as a monolith, you can't choose which API version you want. You choose the OS version. Also, it is not clear how versioning works. I'm afraid we'll have a long stale list of "0.1.0", so just adding more confusion for the developers.

Reply:
Another point, `@since` at an API level doesn't say anything useful to me because an API can change a lot since its initial addition. This should be used per-function, Python docs are a great example.

Reply:
> IMHO, API versions are redundant. Zephyr is used as a monolith, you can't choose which API version you want. You choose the OS version. Also, it is not clear how versioning works. I'm afraid we'll have a long stale list of "0.1.0", so just adding more confusion for the developers.

This is not about having someone select the version, it is about tracking API status and stability and breaking changes. We had this in textual form in a table and we agreeded to move this to the where the API is defined and track it using semantic versioning.

Reply:
To be clear, I am not opposed to adding the `@since` and `@version` tags (I believe this was already agreed upon previously). I am questioning the usability of having all of these in a long list in the documentation (as opposed to listing them for each individual API function in the API documentation).

Reply:
> To be clear, I am not opposed to adding the `@since` and `@version` tags (I believe this was already agreed upon previously). I am questioning the usability of having all of these in a long list in the documentation (as opposed to listing them for each individual API function in the API documentation).

well, they will be in the doxygen API docs and the headers and in sphinx and in other places, this page is generated automatically and we can decide how to display it and where to display any way we want. A long list that is up to date and current is 1000 times better than a short and a stale list.
Btw, another potential usage of this information is diffing the API versions between 2 releases and help identify API changes that can feed into the migration guide etc.